File: Comprehensive_Data_Operations_Simulation/Q3. Subrectangle Queries/Q3_SubrectangleQueries.py
class SubrectangleQueries(object):

    def __init__(self, rectangle):
        """
        :type rectangle: List[List[int]]
        """
        self.rect = rectangle                          # store the matrix

    def updateSubrectangle(self, row1, col1, row2, col2, newValue):
        """
        :type row1: int
        :type col1: int
        :type row2: int
        :type col2: int
        :type newValue: int
        :rtype: None
        """
        for r in range(row1, row2 + 1):                # loop through rows in range
            for c in range(col1, col2 + 1):            # loop through columns in range
                self.rect[r][c] = newValue             # update each cell

    def getValue(self, row, col):
        """
        :type row: int
        :type col: int
        :rtype: int
        """
        return self.rect[row][col]                    # return value at (row, col)

File: Comprehensive_Data_Operations_Simulation/Q3. Subrectangle Queries/Q3_SubrectangleQueries.java
class SubrectangleQueries {

    private int[][] rect;                          // store the rectangle

    public SubrectangleQueries(int[][] rectangle) {
        rect = rectangle;                          // copy the reference to the matrix
    }
    
    public void updateSubrectangle(int row1, int col1, int row2, int col2, int newValue) {

        for (int r = row1; r <= row2; r++) {       // loop through all rows in the given range
            for (int c = col1; c <= col2; c++) {   // loop through all columns in the range
                rect[r][c] = newValue;             // update each cell with newValue
            }
        }
    }
    
    public int getValue(int row, int col) {
        return rect[row][col];                     // simply return the value at (row, col)
    }
}
